packages/@react-spectrum/menu/src/Menu.tsx

```diff
@@ -29,49 +30,56 @@ function Menu<T extends object>(props: SpectrumMenuProps<T>, ref: DOMRef<HTMLULi
   };
 
   let domRef = useDOMRef(ref);
+  let scopedRef = useRef(null);
   let state = useTreeState(completeProps);
   let {menuProps} = useMenu(completeProps, state, domRef);
   let {styleProps} = useStyleProps(completeProps);
   useSyncRef(contextProps, domRef);
 
   return (
-    <ul
-      {...menuProps}
-      {...styleProps}
-      ref={domRef}
-      className={
-        classNames(
-          styles,
-          'spectrum-Menu',
-          styleProps.className
-        )
-      }>
-      {[...state.collection].map(item => {
-        if (item.type === 'section') {
-          return (
-            <MenuSection
-              key={item.key}
-              item={item}
-              state={state}
-              onAction={completeProps.onAction} />
-          );
-        }
+    <MenuStateContext.Provider value={{state, container: scopedRef, menu: domRef}}>
+      <FocusScope contain={state.expandedKeys.size > 0}>
+        <div ref={scopedRef} style={{display: 'inline-flex', overflow: 'hidden'}}>
```

Comment:
Should these styles be inline or come from a class style?

Reply:
doesn't really matter, not a place that people can access through UNSAFE style or classname and isn't about color or sizes, so doesn't need any tokens